plugin/rt_upload_download_cleanup.go

```diff
@@ -0,0 +1,123 @@
+package plugin
+
+import (
+	"fmt"
+	"os"
+	"time"
+)
+
+var UploadCmdJsonTagToExeFlagMapStringItemList = []JsonTagToExeFlagMapStringItem{
+	{"--build-name=", "PLUGIN_BUILD_NAME", false, false},
+	{"--build-number=", "PLUGIN_BUILD_NUMBER", false, false},
+	{"--module=", "PLUGIN_MODULE", false, false},
+	{"--project=", "PLUGIN_PROJECT", false, false},
+	{"--server-id=", "PLUGIN_SERVER_ID", false, false},
+	{"--spec=", "PLUGIN_SPEC", false, false},
+	{"--spec=", "PLUGIN_SPEC_PATH", false, false},
+}
+
+func GetUploadCommandArgs(args Args) ([][]string, error) {
+
+	var cmdList [][]string
+	jfrogConfigAddConfigCommandArgs, err := GetConfigAddConfigCommandArgs("tmpServerId", args.Username, args.Password,
+		args.URL, args.AccessToken, args.APIKey)
+	if err != nil {
+		return cmdList, err
+	}
+
+	if args.Spec != "" {
+		fileName := getTimestampedFileName()
+		err = writeToFile(fileName, args.Spec)
+		if err != nil {
+			return cmdList, err
+		}
+		args.Spec = ""
+		args.SpecPath = fileName
+	}
+
+	uploadCommandArgs := []string{"rt", "upload", args.Source, args.Target}
```

Comment:
this plugin in it's capacity, already supports upload so let's only add download and cleanup as part of this conversion